### PR TITLE
tommy/#236-pixiJS-init

### PR DESCRIPTION
**Pull request by teelau**
:tickets: **Ticket(s)**: Closes #

---

## :construction_worker: Changes

Init connection to server from new client based on pixiJS graphics library, no framework

## :flashlight: Testing Instructions

Explain how to test your changes, if applicable.
`make server`
in the web directory containing index.html:
`python -m SimpleHTTPServer -8000`
`visit localhost:8000`

### CODE REVIEW

**Comment by teelau** [comment]
it would be nice to have a local copy of pixiJS to keep version consistent, how to disable it from being built by travis, its minified.

**Comment by brian-nguyen** [comment]
> it would be nice to have a local copy of pixiJS to keep version consistent
Hey Tommy, no need for this. The CDN allows us to lock the version. Thanks for bringing this up. I should lock down the bootstrap version.

File: client/web/connection.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export default async function connectPlayer() {
+    const response = await fetch(`http://${address}/start`);
+    const res = await response.json();
+
+    // Address of lobby to connect to
+    console.log(res.location);
+
+    if (window.WebSocket) {
+        let socket = new WebSocket(`ws://${address}/connect`);
```
Comment (brian-nguyen) [inline client/web/connection.js]: Did this change from `const` to `let`?

File: client/web/index.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<!doctype html>
+<html>
+<head>
+  <meta charset="utf-8">
+  <title>Hello World</title>
+</head>
+  <script src="pixi.min.js"></script>
```
Comment (brian-nguyen) [inline client/web/index.html]: See my comment on the CDN :)

File: client/web/main.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    rocket.anchor.set(0.5,0.5);
+    rocket.x = 96;
+    rocket.y = 96;
+    rocket.scale.set(0.2, 0.2); //scale of original png object
+    rocket.rotation = 0.5; //radians
+    app.stage.addChild(rocket);
+    connectPlayer();
+    console.log(window.Game)
+}
+
```
Comment (brian-nguyen) [inline client/web/main.js]: This file looks a great start!

File: client/web/main.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import connectPlayer from './connection.js';
+window.Game = {
```
Comment (brian-nguyen) [inline client/web/main.js]: I don't think we should be adding properties to the window object. A standalone object should be fine

File: client/web/connection.js (hunk 1 above)
Comment (teelau) [reply client/web/connection.js]: yes, const is probably better

File: client/web/index.html (hunk 2 above)
Comment (teelau) [reply client/web/index.html]: kk

File: client/web/main.js (hunk 3 above)
Comment (teelau) [reply client/web/main.js]: :)

File: client/web/main.js (hunk 4 above)
Comment (teelau) [reply client/web/main.js]: It's not working for me when i try to make a global variable accessible across files. Another way would be to pass the variable into the functions. Do you know how to make a nice global variable without window object?

Comment (brian-nguyen) [reply client/web/main.js]: Hmm, I'll look into this :)

**Pull request merged**